src/ngResource/resource.js

```diff
@@ -593,7 +601,13 @@ angular.module('ngResource', ['ng']).
               extend({}, extractParams(data, action.params || {}), params),
               action.url);
 
-            var promise = $http(httpConfig).then(function(response) {
+            var promise = $q.resolve(httpConfig);
```

Comment:
Well, that's how my brain addressed the problem. First, peel everything into a promise chain. Second, add the desired step into that chain. While I can collapse it easily enough, I personally find it more legible, as each statement is discrete and easily understood.

I don't mind updating the commit though, if there's consensus that your proposed syntax is more desirable.